pkgs/applications/misc/calibre/default.nix

```diff
@@ -42,9 +43,20 @@ mkDerivation rec {
     # the unrar patch is not from debian
   ] ++ lib.optional (!unrarSupport) ./dont_build_unrar_plugin.patch;
 
+  escaped_pyqt5_dir = builtins.replaceStrings ["/"] ["\\/"] (toString python3Packages.pyqt5);
+  platform_tag =
+    if stdenv.hostPlatform.isDarwin then
+      "\\\"WS_MACX\\\""
+    else if stdenv.hostPlatform.isWindows then
+      "\\\"WS_WIN\\\""
+    else
+      "\\\"WS_X11\\\"";
+
   prePatch = ''
-    sed -i "/pyqt_sip_dir/ s:=.*:= '${pythonPackages.pyqt5}/share/sip/PyQt5':"  \
-      setup/build_environment.py
+    sed -i "s/\[tool.sip.project\]/[tool.sip.project]\nsip-include-dirs = [\"${escaped_pyqt5_dir}\/share\/sip\/PyQt5\"]/g" \
```

Comment:
Maybe we should do a patch for that because it seems quite complicated.

Reply:
I'm not sure how to do that, the patch would be parametrized (e.g., by `pythonPackages.pyqt5`)

Reply:
I think you can do @pythonPackages.pyqt5@ and replace it but I am not fully sure how to do that.

Reply:
I agree a patch is preferred, because substitutes may not happen.